activesupport/lib/active_support/testing/setup_and_teardown.rb

```diff
@@ -44,8 +44,15 @@ def before_setup # :nodoc:
       end
 
       def after_teardown # :nodoc:
-        run_callbacks :teardown
+        begin
+          run_callbacks :teardown
+        rescue => e
+          error = e
```

Comment:
What if multiple `teardown` callbacks would raise?

Reply:
The first one would halt the execution of the others. Like it would happen if we were using `def teardown` instead of the `teardown` callback.